pkg/apply/applier.go

```diff
@@ -322,3 +359,37 @@ func localNamespaces(localInv inventory.InventoryInfo, localObjs []object.ObjMet
 	}
 	return namespaces
 }
+
+func handleValidationError(eventChannel chan<- event.Event, err error) {
+	switch tErr := err.(type) {
+	case *validation.Error:
+		// handle validation error about one or more specific objects
+		eventChannel <- event.Event{
+			Type: event.ValidationType,
+			ValidationEvent: event.ValidationEvent{
+				Identifiers: tErr.Identifiers(),
+				Error:       tErr,
+			},
+		}
+	default:
+		// handle general validation error (no specific object)
+		eventChannel <- event.Event{
+			Type: event.ValidationType,
+			ValidationEvent: event.ValidationEvent{
+				Error: tErr,
+			},
+		}
+	}
+}
+
+// extractInvalidIds extracts IDs from a list of validation errors, ingoring
+// other error types.
+func extractInvalidIds(errs []error) object.ObjMetadataSet {
+	var invalidIds object.ObjMetadataSet
+	for _, err := range errs {
+		if tErr, ok := err.(*validation.Error); ok {
```

Comment:
Why is that better than type comparison?

Reply:
Oh, i see. errors. As will recursively Unwrap errors until it finds the specified type. That makes sense, in case any of the validation errors are wrapped. Unfortunately, it wont work if there's a MultiError in the stack. But in this case, I already unwrapped those before adding them to the Collector (this code is in the Collector now).

Reply:
done